charts/internal/seed-controlplane/charts/csi-driver-controller/templates/_deployment.tpl

```diff
@@ -0,0 +1,199 @@
+{{- define "csi-driver-controller.deployment" -}}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: csi-driver-controller-{{ .role }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: csi
+    role: controller-{{ .role }}
+spec:
+  replicas: {{ .Values.replicas }}
+  revisionHistoryLimit: 0
+  selector:
+    matchLabels:
+      app: csi
+      role: controller-{{ .role }}
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+{{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+      creationTimestamp: null
+      labels:
+        app: csi
+        role: controller-{{ .role }}
+        networking.gardener.cloud/to-dns: allowed
+        networking.gardener.cloud/to-public-networks: allowed
+        networking.gardener.cloud/to-shoot-apiserver: allowed
+    spec:
+      containers:
+      - name: csi-driver
+        image: {{ index .Values.images (print "csi-driver-" .role) }}
+        imagePullPolicy: IfNotPresent
+        args :
+        - --endpoint=$(CSI_ENDPOINT)
+        {{- if eq .role "file" }}
+        - --nodeid=dummy
+        {{- end }}
+        - --v=5
+        env:
+        - name: CSI_ENDPOINT
+          value: unix://{{ .Values.socketPath }}/csi.sock
+        - name: AZURE_CREDENTIAL_FILE
+          value: /etc/kubernetes/cloudprovider/cloudprovider.conf
+{{- if .Values.resources.driver }}
+        resources:
+{{ toYaml .Values.resources.driver | indent 10 }}
+{{- end }}
+        ports:
+        - name: healthz
+          containerPort: 9808
+          protocol: TCP
+        livenessProbe:
+          failureThreshold: 5
+          httpGet:
+            path: /healthz
+            port: healthz
+            scheme: HTTP
+          initialDelaySeconds: 10
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 3
+        volumeMounts:
+        - name: socket-dir
+          mountPath: {{ .Values.socketPath }}
+        - name: cloud-provider-config
+          mountPath: /etc/kubernetes/cloudprovider
+
+      - name: csi-provisioner
+        image: {{ index .Values.images "csi-provisioner" }}
+        imagePullPolicy: IfNotPresent
+        args:
+        - --connection-timeout=15s
```

Comment:
Sorry, `connection-timeout` is deprecated and has no effect since v1.1.0. Ref https://github.com/kubernetes-csi/external-provisioner/blob/master/CHANGELOG-1.1.md.
```suggestion
```